src/Workspaces/Core/Portable/FindSymbols/DeclaredSymbolInfo.cs

```diff
@@ -98,26 +126,32 @@ internal enum DeclaredSymbolInfoKind : byte
             ImmutableArray<string> inheritanceNames,
             bool isNestedType = false, int parameterCount = 0, int typeParameterCount = 0)
         {
-            Name = Intern(stringTable, name);
-            NameSuffix = Intern(stringTable, nameSuffix);
-            ContainerDisplayName = Intern(stringTable, containerDisplayName);
-            FullyQualifiedContainerName = Intern(stringTable, fullyQualifiedContainerName);
-            Span = span;
-            InheritanceNames = inheritanceNames;
-
             const uint MaxFlagValue = 0b1111;
+
             Contract.ThrowIfTrue((uint)accessibility > MaxFlagValue);
             Contract.ThrowIfTrue((uint)kind > MaxFlagValue);
+
             parameterCount = Math.Min(parameterCount, (byte)MaxFlagValue);
             typeParameterCount = Math.Min(typeParameterCount, (byte)MaxFlagValue);
 
-            _flags =
+            var flags =
                 (uint)kind |
                 ((uint)accessibility << 4) |
                 ((uint)parameterCount << 8) |
                 ((uint)typeParameterCount << 12) |
                 ((isNestedType ? 1u : 0u) << 16) |
                 ((isPartial ? 1u : 0u) << 17);
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            return new DeclaredSymbolInfo(
+                Intern(stringTable, name),
+                Intern(stringTable, nameSuffix),
+                Intern(stringTable, containerDisplayName),
+                Intern(stringTable, fullyQualifiedContainerName),
```

Comment:
❔ Do we not want to move this into the constructor?

Reply:
no.  i wanted to be able to use thsi as a DataContract.  so the pure data part is in the tyep/constructor.  but the types that create tehse on the OOP side go through teh factory method to share stirngs there.